### PR TITLE
network: add UI to manage server certificates

### PR DESCRIPTION
**Pull request by thc202**
Add UI options to:
 - Configure the validity of the root CA certificate and issued
 certificates.
 - Generate, import, save, and view the root CA certificate.

Moves core deprecated functionality, zaproxy/zaproxy#6950.
Move and update core help, zaproxy/zap-core-help#393.

### CODE REVIEW

File: addOns/network/src/main/javahelp/help/contents/options/servercertificates.html [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+<HEAD>
+<META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=utf-8">
+<TITLE>Server Certificates</TITLE>
+</HEAD>
+<BODY>
+	<H1>Server Certificates</H1>
+	This screens allows to manage and configure the root CA certificate and issued certificates.
+
+	<p>
+	OWASP ZAP allows you to transparently decrypt SSL connections.
```
Comment (kingthorin) [inline addOns/network/src/main/javahelp/help/contents/options/servercertificates.html:12]: Should we update this to include TLS?

File: addOns/network/src/main/javahelp/help/contents/options/servercertificates.html [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+	That's why, the only way to decrypt or intercept the transmission,
+	is to do a 'manipulator in the middle' approach.
+	</p>
+
+	<H2>Overview</H2>
+	<p>
+		<img alt="manipulator in the middle" src="../images/manipulatorinthemiddle.png">
+	</p>
+	<p>
+		In short words, every data send to and received from the server
```
Comment (kingthorin) [inline addOns/network/src/main/javahelp/help/contents/options/servercertificates.html:25]:
```suggestion
		In short words, all data sent to and received from the server
```

File: addOns/network/src/main/javahelp/help/contents/options/servercertificates.html (hunk 1 above)
Comment (kingthorin) [reply addOns/network/src/main/javahelp/help/contents/options/servercertificates.html:12]: Throughout… 😉

**Review by kingthorin — COMMENTED** [review]
Is the help text new or just migrated?

I think I need to do more of a review of it in my laptop, cell phone isn’t really cutting it.

However if it’s just being migrated I can tackle changes/updates in another PR

**Comment by thc202** [comment]
> Is the help text new or just migrated?

Migrated with some changes (e.g. match the new labels, the validity):
```diff
10c11
< ZAP Root CA certificate
---
> Root CA certificate
24c25
< Every generated Root CA certificate is valid for one year. After that period you have to create a new one.
---
> The generated Root CA certificate is valid, by default, for one year. After that period you have to create a new one.
81c82
< And yes, that example will work - its the Superfish certificate!
---
> And yes, that example will work - it's the Superfish certificate!
84,85c85,86
< In the options dialog of ZAP you're seeing the raw bytes (hexa-decimal encoded) of the certificate. The option "view" tries to use your system's default viewing tool for ".CER" files. On Windows, this is typically the same, when exporting the certificate and double clicking on it.
< Save/Export
---
> In the options dialog of ZAP you're seeing the certificate in PEM format. The option "view" tries to use your system's default viewing tool for ".CER" files. On Windows, this is typically the same, when exporting the certificate and double clicking on it.
> Save
88c89
< Dynamic certificates
---
> Issued Certificates
92,94c93,95
< Every dynamically generated certificate is valid for 1000 days.
< Every dynamically generated certificate is 2048 bit strong (RSA with SHA1).
< Every dynamically generated certificate has a random serial number. Every dynamically generated certificate consists of the following identifiers:
---
> The issued certificates are valid, by default, for 368 days.
> The issued certificates is 2048 bit strong (RSA with SHA1).
> The issued certificates has a random serial number. The issued certificates consists of the following identifiers:
102c103
< Side note: Each time you start ZAP, internally a random serial number offset is generated. Every dynamically generated certificate will use this offset plus an increasing counter. For example, first dynamically certificate has serial number 2314, the second one 2315, the third one 2316 and so on. The reason for this is simple: browsers are also caching certificates. When you restart ZAP but don't restart your browser, it could happen, that the browser sees the same certificate but with different serial number. In the end, the browser would complain about and reject the certificate. By using the random offset (internally 48bit random number), the chances are 1 to 281.474.976.710.656 that when restarting ZAP, the serial number offset is a different one.
---
> Side note: Each time you start ZAP, internally a random serial number offset is generated. The issued certificates will use this offset plus an increasing counter. For example, first the issued certificate has serial number 2314, the second one 2315, the third one 2316 and so on. The reason for this is simple: browsers are also caching certificates. When you restart ZAP but don't restart your browser, it could happen, that the browser sees the same certificate but with different serial number. In the end, the browser would complain about and reject the certificate. By using the random offset (internally 48bit random number), the chances are 1 to 281.474.976.710.656 that when restarting ZAP, the serial number offset is a different one.
109c110
< The easiest way is to click on view and choose 'Install certificate'. Alternatively, you can save/export your generated certificate (copy it to you target computer) and double click the .CER file. When doing so, the regular Windows wizard for certificate installation assistance is poping up. In this wizard manually choose the certificate store. Do NOT let Windows choose automatically the certificate store. Choose 'trusted root certificates' as store and finalize the wizard.
---
> The easiest way is to click on view and choose 'Install certificate'. Alternatively, you can save/export your generated certificate (copy it to you target computer) and double click the .CER file. When doing so, the regular Windows wizard for certificate installation assistance is popping up. In this wizard manually choose the certificate store. Do NOT let Windows choose automatically the certificate store. Choose 'trusted root certificates' as store and finalize the wizard.
```

**Comment by kingthorin** [comment]
That’s pretty minor. Okay let’s leave it as-is for this PR, I can review it and tackle changes separately.

**Review by kingthorin — APPROVED** [review]

**Review by ricekot — APPROVED** [review]
Thank you!

**Pull request merged**